ftw/linkchecker/plone_site.py

```diff
@@ -0,0 +1,47 @@
+from AccessControl.SecurityManagement import newSecurityManager
+from Acquisition import aq_parent
+from ftw.linkchecker import LOGGER_NAME
+from zope.component.hooks import setSite
+from zope.globalrequest import setRequest
+import AccessControl
+import logging
+
+
+class PloneSite():
+
+    def __init__(self, plone_site, configuration):
+        self.configuration = configuration
+        self.upload_path = ''
+        self.logger = logging.getLogger(LOGGER_NAME)
+
+        self._setup_plone(plone_site)
+        self.obj = plone_site
+        self.configuration = self._get_config_for_portal()
+        if self.configuration:
+            self._set_upload_path()
+
+    def _setup_plone(self, plone_site_obj):
+        setRequest(aq_parent(plone_site_obj).REQUEST)
+        user = AccessControl.SecurityManagement.SpecialUsers.system
+        user = user.__of__(plone_site_obj.acl_users)
+        newSecurityManager(plone_site_obj, user)
+        setSite(plone_site_obj)
+
+    def _get_config_for_portal(self):
+        try:
+            site_config = [site_config for site_config
+                           in self.configuration.sites
+                           if site_config.site_name == '/'.join(
+                               self.obj.getPhysicalPath())][0]
```

Comment:
You could use `next` here:
https://stackoverflow.com/questions/7006839/alternative-to-list-comprehension-if-there-will-be-only-one-result?answertab=active#tab-top

Reply:
I created an issue https://github.com/4teamwork/ftw.linkchecker/issues/84 to do that later on.